src/api/aux.cpp

```diff
@@ -19,72 +19,75 @@ void parse_json_file(std::vector<Token> &tokens, std::vector<Order> &orders, std
     // TODO: Make input reading safe, i.e., handle all potential exceptions/errors
 
     // For-loop that reads all TOKEN information
-    
-    for (auto &i: json_file["tokens"].items()) {
-        boost::multiprecision::mpf_float t;
-        if (i.value()["external_price"].is_null())
-            t = -1;
-        else {
-            //std::string tt = std::to_string(i.value()["external_price"]);
-            std::string tt = (i.value()["external_price"]).dump();
-            t = static_cast<boost::multiprecision::mpf_float>(tt);
+    for (auto &tkn: json_file["tokens"].items()) {
+        boost::multiprecision::mpf_float external_price;
```

Comment:
tip: do something like
```c++
typedef boost::multiprecision::mpf_float BigFloat;
typedef boost::multiprecision::mpz_int BigInt;
```

somewhere at the top of the file (or in a file you always include), and then use BigFloat/BigInt instead of the larger type identifiers.

Reply:
I generally avoid too many definitions as I think that then it gets harder to actually keep track of the definitions themselves. Of course, so far, I haven't used any, so I will follow your advice!